PlagiarismPlugin.inc.php

```diff
@@ -43,169 +56,291 @@ public function getDescription() {
 	}
 
 	/**
-	 * Send submission files to iThenticate.
-	 * @param $hookName string
-	 * @param $args array
+	 * Setup the handler for webhook request
+	 * 
+	 * @param string $hookName `LoadComponentHandler`
+	 * @param array $params
+	 * 
+	 * @return bool
 	 */
-	public function callback($hookName, $args) {
-		$request = Application::get()->getRequest();
-		$context = $request->getContext();
-		$contextPath = $context->getPath();
-		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
-		$submission = $submissionDao->getById($request->getUserVar('submissionId'));
-		$publication = $submission->getCurrentPublication();
+	public function setupWebhookHandler($hookName, $params) {
+		$component =& $params[0];
 
-		require_once(dirname(__FILE__) . '/vendor/autoload.php');
+		if ($component == 'plugins.generic.plagiarism.controllers.PlagiarismWebhookHandler') {
+			import($component);
+			PlagiarismWebhookHandler::setPlugin($this);
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Confirm EULA, create submission and upload submission files to iThenticate service
+	 * 
+	 * @param string $hookName `submissionsubmitstep4form::execute`
+	 * @param array $args
+	 * 
+	 * @return bool
+	 */
+	public function submitForPlagiarismCheck($hookName, $args) {
+		$request = Application::get()->getRequest(); /** @var Request $request */
+		$form =& $args[0]; /** @var SubmissionSubmitStep4Form $form */
+		$submission = $form->submission; /** @var Submission $submission */
+		$context = $request->getContext(); /** @var Context $context */
+		$publication = $submission->getCurrentPublication(); /** @var Publication $publication */
+		$author = $publication->getPrimaryAuthor(); /** @var Author $author */
 
 		// try to get credentials for current context otherwise use default config
-        	$contextId = $context->getId();
-		list($username, $password) = $this->getForcedCredentials();
-		if (empty($username) || empty($password)) {
-			$username = $this->getSetting($contextId, 'ithenticateUser');
-			$password = $this->getSetting($contextId, 'ithenticatePass');
-		}
+		list($apiUrl, $apiKey) = $this->hasForcedCredentials()
+			? $this->getForcedCredentials()
+			: [
+				$this->getSetting($context->getId(), 'ithenticateApiUrl'), 
+				$this->getSetting($context->getId(), 'ithenticateApiKey')
+			];
 
-		$ithenticate = null;
-		try {
-			$ithenticate = new \bsobbe\ithenticate\Ithenticate($username, $password);
-		} catch (Exception $e) {
-			$this->sendErrorMessage($submission->getId(), $e->getMessage());
+		import("plugins.generic.plagiarism.IThenticate");
+
+		/** @var \IThenticate $ithenticate */
+		$ithenticate = new IThenticate(
+			$apiUrl,
+			$apiKey,
+			static::PLUGIN_INTEGRATION_NAME,
+			$this->getCurrentVersion()->getData('current')
+		);
+
+		// Try to validate and confirm the EULA on user's behalf
+		if (!$ithenticate->confirmEula($request->getUser(), $context)) {
```

Comment:
Does the EULA need to be confirmed via the API before the submission can be received by iThenticate?

Reply:
Yes but as per our discussion, we are not stamping this to submitter to continuous confirmation and polling .